### DISCUSSION TITLE
Using Material Web in HTML/Express.js

### DISCUSSION THREAD

**Discussion by SlimyFrog123** [material-components/material-web#5264] [Q&A]
How could I use Material Web in a plain HTML document or an [Express.js](https://expressjs.com/) app? I have tried [UNPKG](https://unpkg.com/browse/@material/web@1.0.1/list/), but that did not seem to work properly. But then again, I don't really have any experience with UNPKG or projects like this, where you must include the components via JavaScript.

My apologies if this is an amateur question or cannot be done.

Thanks in advance!

**Comment by CornerSyrup** [comment]
@material/web is a module package and require bundling, Unpkg only serve script as is, try esm.run by jsdelivr.

```html
<script type="module" src="https://esm.run/@material/web@1.x.x/all.js"></script>
```

**Reply by vdegenne** [reply]
You can tell unpkg to serve module files as well.
```html
<script type="module" src="https://www.unpkg.com/@material/web@1.0.1/all.js?module"></script>
```
(@SlimyFrog123)

**Reply by jrmanrique** [reply]
Requesting via `unpkg` with the `?module` flag seems to raise `Failed to load resource: the server responded with a status of 500 ()` errors with `lit-element` and `@lit`.

**Comment by Aworldc** [comment]
1. Install material web and esbuild:
     `npm i @material/web esbuild`
2. create a js file, call it what you want, for example, `src/components.js`
3. in that file, import the components you want to use, for example:
    ```js
    import '@material/web/button/filled-button.js'
    import '@material/web/button/outlined-button.js'
    ```
4. use esbuild to bundle that js file:
    ```bash
    $ npx esbuild --bundle --minify --outfile=bundle.js src/components.js
    ```
    this is assuming the js file you created earlier is at `src/components.js` and the file the output should go to is bundle.js

5. create an express route that serves the bundle.js file created in the previous step. something like
    ```js
        app.get('/bundle.js', (req, res) => {
            res.sendFile(path.join(_dirname, 'bundle.js'))
        }
    ```
6. load that js file in your web page:
    ```html
        <script src="bundle.js">
    ```

**Comment by treeder** [comment]
See this answer: https://github.com/material-components/material-web/discussions/5239#discussioncomment-7831026
